File: Python/textbook_track/chapter03/ba3a.py
# Generate the k-mer Composition of a String
import sys


def calc_k_mer_composition(text, k):
    composition = []
    for ix in range(len(text)-k+1):
        k_mer = text[ix:ix+k]
        composition.append(k_mer)
    return sorted(composition)


def main():
    data = sys.stdin.read().splitlines()
    k = int(data[0])
    text = data[1]
    result = calc_k_mer_composition(text, k)
    for k_mer in result:
        print(k_mer)


if __name__ == '__main__':
    main()

File: Julia/textbook_track/chapter03/ba3a.jl
# Generate the k-mer Composition of a String

function calc_k_mer_composition(text, k)
    composition = String[]
    for ix in 1:(length(text)-k+1)
        substring = text[ix:ix+k-1]
        push!(composition, substring)
    end
    sort(composition)
end


function main()
    k = parse(Int64, readline())
    text = readline()
    result = calc_k_mer_composition(text, k)
    foreach(println, result)
end


main()
